src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -21,6 +21,9 @@ namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.DeclareAsNullable
     [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.DeclareAsNullable), Shared]
     internal class CSharpDeclareAsNullableCodeFixProvider : SyntaxEditorBasedCodeFixProvider
     {
+        private const string IsConditionalOperatorEquivalenceKey = nameof(IsConditionalOperatorEquivalenceKey);
+        private const string IsOtherEquivalenceKey = nameof(IsOtherEquivalenceKey);
```

Comment:
i'd prefer different subclasses.  

if we're just going to report differenet issues and have to filter them anyways, using a single type that does this and has to do this stuff is just tryin to combine things only to do more work.

Reply:
note: it's also unclear to me why we want to segment the fix-alls.  As far as the user is concerned, it's the same diagnostic.  Why not have fix-all just fix all of them?

Reply:
> it's also unclear to me why we want to segment the fix-alls. As far as the user is concerned, it's the same diagnostic. Why not have fix-all just fix all of them?

It was requested in issue description by @sharwell.

> 📝 The code fix should use a unique Equivalence Key for this situation.

Details, why this was requested, are unclear to me as well.

Reply:
Gotcha.  Will wait on @sharwell for clarification here.  Just to check, but are these the same diagnostic?  Or is there a separate diagnostic for each case?

personally, i would want a fix all for each unique *diagnostic* code, not for the different constructs producing this.

Reply:
> Just to check, but are these the same diagnostic? Or is there a separate diagnostic for each case?

As far as I understand all are the same diagnostic (CS8603) and the codefix is same for all of them. Just to change return type to be nullable. So they share the same implementation. I hope I've made it a little bit more clear 🙂 .

Reply:
@sharwell Can you clarify the request for distinct fix alls? I'm not sure why we would want them either, especially because it seems like the covering of a bunch of separate special cases doesn't seem right...

Reply:
Certainly. #39422 is a case where CS8603 is reported even though the user isn't _explicitly_ returning null. The distinction allows users to scrutinize cases where null might be accidentally returned, separately from cases where it is explicitly returned.